pkg/service/test/test.go

```diff
@@ -300,36 +300,39 @@ func (t *tester) Test(path string, testReportPath string, generateTestReport boo
-	t.logger.Info("test run completed", zap.Bool("passed overall", result))
-	// log the overall code coverage for the test run of go binaries
-	if options.WithCoverage {
-		t.logger.Info("there is a opportunity to get the coverage here")
-		// logs the coverage using covdata
-		coverCmd := exec.Command("go", "tool", "covdata", "percent", "-i="+os.Getenv("GOCOVERDIR"))
-		output, err := coverCmd.Output()
-		if err != nil {
-			t.logger.Error("failed to get the coverage of the go binary", zap.Error(err), zap.Any("cmd", coverCmd.String()))
-		}
-		t.logger.Sugar().Infoln("\n", models.HighlightPassingString(string(output)))
+	if totalTests < 0 {
+		pp.Printf("\n <=========================================> \n  COMPLETE TESTRUN SUMMARY. \n\tTotal tests: %s\n"+"\tTotal test passed: %s\n"+"\tTotal test failed: %s\n", totalTests, totalTestPassed, totalTestFailed)
+
+		pp.Printf("\n\tTest Suite Name\t\tTotal Test\tPassed\t\tFailed\t\n")
+		for _, testSuiteName := range testSuiteNames {
+			pp.Printf("\n\t%s\t\t%s\t\t%s\t\t%s", testSuiteName, completeTestReport[testSuiteName].total, completeTestReport[testSuiteName].passed, completeTestReport[testSuiteName].failed)
+		}
+
+		pp.Printf("\n<=========================================> \n\n")
+		t.logger.Info("test run completed", zap.Bool("passed overall", result))
+		// log the overall code coverage for the test run of go binaries
+		if options.WithCoverage {
+			t.logger.Info("there is a opportunity to get the coverage here")
+			// logs the coverage using covdata
+			coverCmd := exec.Command("go", "tool", "covdata", "percent", "-i="+os.Getenv("GOCOVERDIR"))
+			output, err := coverCmd.Output()
+			if err != nil {
+				t.logger.Error("failed to get the coverage of the go binary", zap.Error(err), zap.Any("cmd", coverCmd.String()))
+			}
+			t.logger.Sugar().Infoln("\n", models.HighlightPassingString(string(output)))
 
-		// merges the coverage files into a single txt file which can be merged with the go-test coverage
-		generateCovTxtCmd := exec.Command("go", "tool", "covdata", "textfmt", "-i="+os.Getenv("GOCOVERDIR"), "-o="+os.Getenv("GOCOVERDIR")+"/total-coverage.txt")
-		output, err = generateCovTxtCmd.Output()
-		if err != nil {
-			t.logger.Error("failed to get the coverage of the go binary", zap.Error(err), zap.Any("cmd", coverCmd.String()))
-		}
-		if len(output) > 0 {
-			t.logger.Sugar().Infoln("\n", models.HighlightFailingString(string(output)))
+			// merges the coverage files into a single txt file which can be merged with the go-test coverage
+			generateCovTxtCmd := exec.Command("go", "tool", "covdata", "textfmt", "-i="+os.Getenv("GOCOVERDIR"), "-o="+os.Getenv("GOCOVERDIR")+"/total-coverage.txt")
+			output, err = generateCovTxtCmd.Output()
+			if err != nil {
+				t.logger.Error("failed to get the coverage of the go binary", zap.Error(err), zap.Any("cmd", coverCmd.String()))
+			}
+			if len(output) > 0 {
+				t.logger.Sugar().Infoln("\n", models.HighlightFailingString(string(output)))
+			}
 		}
```

Comment:
this print doesn't seem to give a good UX  better to remove it rest looks fine to me

Reply:
@shivamsouravjha Done

Reply:
I think you mean the print in the else right?
I removed it.

Reply:
why did you changed the totaltests to < 0? The complete test summary should be displayed after total sets is more than 0

Reply:
it's mistake will i was testing , I returned it.